ports/winit/app.rs

```diff
@@ -183,14 +337,45 @@ impl App {
         }
     }
 
+    /// Takes any outstanding events from the [Minibrowser], converting them to [EmbedderEvent] and
+    /// routing those to the App event queue.
+    fn queue_embedder_events_for_minibrowser_events(&self, minibrowser: &mut RefMut<Minibrowser>) {
+        for event in minibrowser.event_queue.borrow_mut().drain(..) {
+            match event {
+                MinibrowserEvent::Go => {
+                    let browser_id = self.browser.borrow().browser_id().unwrap();
+                    let location = minibrowser.location.borrow();
+                    let Ok(url) = ServoUrl::parse(&location) else {
+                        warn!("failed to parse location");
+                        break;
+                    };
+                    self.event_queue.borrow_mut().push(EmbedderEvent::LoadUrl(browser_id, url));
+                },
+            }
+        }
+    }
+
+    /// Updates the location field when the [Browser] says it has changed, unless the user has
+    /// started editing it without clicking Go.
+    fn update_location_in_toolbar(&self, minibrowser: &mut RefMut<Minibrowser>) -> bool {
+        if !minibrowser.location_dirty.get() {
+            if let Some(current_url) = self.browser.borrow_mut().take_top_level_url_change() {
+                minibrowser.location = RefCell::new(ServoUrl::into_string(current_url));
+                return true;
+            }
+        }
```

Comment:
If you take my suggestion below, I think this could be a function on the Minibrowser struct itself.

Reply:
done in 06e0d2987e42a1fd542a34697be3f7ca37ecdc62